internals/models.py

```diff
@@ -1740,6 +1740,36 @@ def remove_as_component_owner(self, component_name):
         component_name, remove_as_owner=True)
 
 
+class APIOwner(DictModel):
+  """Describes the properties to store raw API_OWNERS content."""
+  url = ndb.StringProperty()
+  raw_content = ndb.TextProperty()
+  created_on = ndb.DateTimeProperty(auto_now_add=True)
+
+  def put(self):
+    """Add API owner content in ndb and delete all other entities."""
+    # Delete all other entities.
+    ndb.delete_multi(self.query().fetch(keys_only=True))
```

Comment:
Deleting all other instances works so long as the API Owners file is the only OWNERS file that we have, but in the future we might have other approvals that parse other OWNERS files from different URLs.   Deleting unneeded entities could still be done like:
`ndb.delete_multi(self.query(OwnersFile.url == url).fetch(keys_only=True))`